packages/react-components/src/TextArea.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright 2017-2020 @polkadot/react-components authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { TextareaHTMLAttributes, useCallback } from 'react';
+import styled from 'styled-components';
+
+import { Labelled } from '.';
+import { ThemeProps } from './types';
+
+interface Props {
+  children?: React.ReactNode;
+  className?: string;
+  help?: React.ReactNode;
+  isError?: boolean;
+  isReadOnly?: boolean;
+  label?: React.ReactNode;
+  onChange?: (arg: string) => void;
+  seed?: string;
+  withLabel?: boolean;
+}
+
+interface TextInputProps extends TextareaHTMLAttributes<any>{
+  withError?: boolean;
+}
+
+const TextInput = ({ withError, ...textAreaProps }: TextInputProps) => <textarea
+  {...textAreaProps}
+  className={[textAreaProps.className, withError ? 'ui-textArea-withError' : ''].join(' ')}
+/>;
+
+const TextAreaInput = styled(TextInput)(({ theme }: ThemeProps) => `
+  background: ${theme.bgInput};
+  border-radius: 4px;
+  border: 1px solid #DDE1EB;
+  box-sizing: border-box;
+  color: ${theme.color};
+  display: block;
+  font-size: 1.15rem;
+  outline: none;
+  padding: 1.75rem 3rem 0.75rem 1.5rem;
+  resize: none;
+  width: 100%;
+
+  &:read-only {
+    background: ${theme.bgInverse};
+    box-shadow: none;
+    outline: none;
+  }
+
+  &.ui-textArea-withError {
+    background: ${theme.bgInputError};
+    color: ${theme.colorError};
+  }
+`);
+
+interface TextAreaWithDropdownProps {
+  children?: React.ReactNode;
+  isError?: boolean;
+  isReadOnly?: boolean;
+  rowsCount?: number;
+  onChange?: (value: string) => void;
+  value?: string;
+  id: string;
+}
+
+function TextAreaWithDropdown ({ children, id, isError, isReadOnly, onChange, rowsCount, value }: TextAreaWithDropdownProps): React.ReactElement<TextAreaWithDropdownProps> {
+  const _onChange = useCallback(
+    ({ target: { value } }: React.ChangeEvent<HTMLTextAreaElement>): void => {
+      onChange && onChange(value);
+    },
+    [onChange]
+  );
+
+  return (
+    <>
+      <TextAreaInput
+        autoCapitalize='off'
+        autoCorrect='off'
+        autoFocus={false}
+        id={id}
+        onChange={_onChange}
+        readOnly={isReadOnly}
+        rows={rowsCount || 2}
+        spellCheck={false}
+        value={value}
+        withError={isError}
+      />
+      {children}
+    </>
+  );
+}
+
+function TextArea ({ children, className, help, isError, isReadOnly, label, onChange, seed, withLabel }: Props): React.ReactElement<Props> {
+  return (
+    <Labelled
+      className={className}
+      help={help}
+      label={label}
+      withLabel={withLabel}
+    >
+      <div className='TextAreaWithDropdown'>
+        <TextAreaWithDropdown
```

Comment:
Really not sure why we have these amount of components - we can just flatten into a single? I struggle to follow this with the 3 layers as it stands.

Reply:
OK, done. I'm pretty sure it will make it harder to reuse TextArea in other places, but for now you're right, it simplifies the code greatly.

Reply:
We really should just use the `TextArea` directly from react-components. I'm sure there are some gaps, but will see them when using.